feat(apigateway): dualstack REST API (#34114)
aws/aws-cdk: packages/aws-cdk-lib/aws-apigateway/README.md

});
```

You can also configure [endpoint IP address type](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-ip-address-type.html).
The default value is `IpAddressType.DUAL_STACK` for private API, and `IpAddressType.IPV4` for regional and edge-optimized API.

```ts
const api = new apigateway.RestApi(this, 'api', {
  endpointConfiguration: {
    types: [ apigateway.EndpointType.REGIONAL ],

>     addressTypes: apigateway.IpAddressType.DUAL_STACK,

Reviewer: ```suggestion
    ipAddressType: apigateway.IpAddressType.DUAL_STACK,
```

I think this will solve the build error

  }
});
```

> **Note**: The private API can only have a `DUAL_STACK` IP address type.

Reviewer: ```suggestion
**Note**: If creating a private API, the `IPV4` IP address type is not supported.
```
